pyanaconda/modules/common/structures/secret.py

```diff
@@ -0,0 +1,168 @@
+#
+# DBus structures for secret data.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import copy
+
+from dasbus.structure import DBusData, get_fields, generate_string_from_data
+from dasbus.typing import *  # pylint: disable=wildcard-import
+
+from pyanaconda.anaconda_loggers import get_module_logger
+from pyanaconda.core.constants import SECRET_TYPE_NONE, SECRET_TYPE_TEXT, SECRET_TYPE_HIDDEN
+
+log = get_module_logger(__name__)
+
+__all__ = ["get_public_copy", "hide_secrets", "SecretData", "SecretDataList"]
+
+
+def get_public_copy(obj: DBusData):
```

Comment:
So if I understand it correctly, I can call `get_public_copy()` on an arbitrary DBus structure & it will automatically handle any nested `SecretData` in the DBus structure, right ?

If this is correct, I think it should be mentioned in the docstring, as at the moment I don't think it is really clear this is possible (without reading the implementation) & that the method actually works on any DBus structure, not just SecretData instances.